openedx/core/djangoapps/oauth_dispatch/jwt.py

```diff
@@ -0,0 +1,190 @@
+"""Utilities for working with ID tokens."""
+import json
+from time import time
+
+from django.conf import settings
+from jwkest import jwk
+from jwkest.jws import JWS
+
+from edx_django_utils.monitoring import set_custom_metric
+from openedx.core.djangoapps.oauth_dispatch.toggles import ENFORCE_JWT_SCOPES
+from student.models import UserProfile, anonymous_id_for_user
+
+
+def create_api_client_jwt(user, secret=None, aud=None, additional_claims=None):
+    """
+    To create JWTs for API clients that are making REST API calls to other edX services.
```

Comment:
It would be great if we could add additional comments to better differentiate between `create_api_client_jwt` and `create_app_access_jwt` in both its current and future usage.  Are we hoping this one eventually goes away (deprecated), or do we plan to keep this around forever?

Reply:
What specifically are you looking for here?  I do not expect this to be deprecated.  This is an attempt at creating factory-like methods for the different types of callers that need to create jwt tokens.  We can iterate on this further as we continue to clean up authn.

Reply:
Note: this is not a blocker.  I do however want to have more clarity for my own sake.  I imagine this will probably require a conversation, but I'll try to clarify some of my thoughts/questions.

First, if I look at the implementations of the two methods I mentioned, and imagine that all the deprecated arguments have been removed, the main differences would be:
1. They each use a different expiration.
2. Scopes (and filters) are only available for the second of these methods.
When I read the summary comment, it doesn't make me think that these would be the differences, but probably I'm not getting something I should be getting.

Here are the two comments for these two methods:
```
To create JWTs for API clients that are making REST API calls to other edX services.
```
```
To create JWTs for OAuth applications needing access to various edX services.
```
I just need some help understanding the difference here.  First, I guess I'm not clear on the definitions of "API clients" and "OAuth applications" and how those are different.  Next, I don't understand why "REST API calls" wouldn't be made in both cases.

I'm just trying to add some meat to my original comment.  Let me know if this clarifies my confusion.

Reply:
This is only the 1st draft of this interface.  I do expect to iterate further on this as we figure things out.

Reply:
Understood. Just trying to use this as a possible learning opportunity as we go along.